### ISSUE TITLE
any way to use patterns for a path matcher?

### ISSUE THREAD

**Issue by cameronbraid** [pimterry/mockttp#9]
I would like to be able to use path patterns like 
`
.get("/**")
`
or 
`
.get("/?**)
`
or most flexible :

```
.match(context=>{
// custom matcher here which can access headers, query string parameters, url from the context

})
```

would this be hard to add ?

**Comment by pimterry** [comment]
Right now there's no way to do that, no. Good idea though!

The former should be fairly easy, though I'd probably lean towards regexes instead. E.g. `.get(/.*/)`. That lets you differentiate by type, so the * in the string isn't ambiguous (does it match a pattern, or a literal `*` only?).

The latter is trickier, but possible too. The problem there is that for browser support you need a live connection from the mock to the browser, which it can use to run the matching code. That can come with latency too. There are similar features I'm planning anyway (mainly for dynamic responses, rather than request matching), but the former option should definitely come first.

Interested in working on a PR for that? I'll get to it eventually, but if you're keen to use it I'd happily accept a PR adding it yourself.

**Comment by pimterry** [comment]
@cameronbraid this is now implemented on master, release coming soon. Is this what you were looking for?

https://github.com/pimterry/mockttp/blob/04fe22edb8c69c33e227210609171f21057c1aaa/test/integration/matchers/method-path-matching.spec.ts#L44-L50

**Comment by pimterry** [comment]
Excellent. Now fixed & published as Mockttp 0.7.2, so I'm going to close this. Try it out, let me know how it goes! Thanks for the suggestion :+1:

**Comment by chrisribe** [comment]
Hello,

I was looking at allowing different RegEx patterns in the url matching.
[RegEx test cases](https://github.com/httptoolkit/mockttp/blob/04fe22edb8c69c33e227210609171f21057c1aaa/test/integration/matchers/method-path-matching.spec.ts)

I see in the test cases you only run tests with direct regex expressions and not strings.
`await server.get(/.*.txt/).thenReply(200, 'Fake file');`

Tried to create a regex pattern via new RegExp("some regex pattern string") but I am having match issues.
```
const regExPattern = new RegExp(regExPatternString, 'g');
await server.get(regExPattern).thenReply(200, 'Fake file');
```
Any tips ?
